### ISSUE TITLE
Errors in CRAN Package Check Results

### ISSUE THREAD

**Issue by levenc** [levenc/posologyr#54]
```r
* using log directory ‘/data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck’
* using R Under development (unstable) (2024-09-02 r87090)
* using platform: x86_64-pc-linux-gnu
* R was compiled by
    gcc-14 (GCC) 14.2.0
    GNU Fortran (GCC) 14.2.0
* running under: Fedora Linux 36 (Workstation Edition)
* using session charset: UTF-8
* using option ‘--no-stop-on-test-error’
* checking for file ‘posologyr/DESCRIPTION’ ... OK
* this is package ‘posologyr’ version ‘1.2.6’
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘posologyr’ can be installed ... [10s/14s] OK
* checking package directory ... OK
* checking ‘build’ directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking code files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking whether startup messages can be suppressed ... OK
* checking use of S3 registration ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... [20s/23s] OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking installed files from ‘inst/doc’ ... OK
* checking files in ‘vignettes’ ... OK
* checking examples ... [34s/33s] OK
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’ [306s/243s]
 [307s/243s] OK
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes ... OK
* checking re-building of vignette outputs ... [550s/434s] ERROR
Error(s) in re-building vignettes:
--- re-building ‘a_posteriori_dosing.Rmd’ using rmarkdown
** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/a_posteriori_dosing_files/figure-html/map_plot_tdm-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 11821 bytes
Input file size = 11911 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 10239
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 10239

Output IDAT size = 10239 bytes (1582 bytes decrease)
Output file size = 10317 bytes (1594 bytes = 13.38% decrease)

malloc(): unsorted double linked list corrupted
--- re-building ‘a_priori_dosing.Rmd’ using rmarkdown
** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/a_priori_dosing_files/figure-html/unnamed-chunk-3-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 11565 bytes
Input file size = 11655 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 10049
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 10049

Output IDAT size = 10049 bytes (1516 bytes decrease)
Output file size = 10127 bytes (1528 bytes = 13.11% decrease)

** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/a_priori_dosing_files/figure-html/prior_plot-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 15575 bytes
Input file size = 15665 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 12809
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 12809

Output IDAT size = 12809 bytes (2766 bytes decrease)
Output file size = 12887 bytes (2778 bytes = 17.73% decrease)

--- finished re-building ‘a_priori_dosing.Rmd’

--- re-building ‘auc_based_dosing.Rmd’ using rmarkdown
** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/auc_based_dosing_files/figure-html/unnamed-chunk-2-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 13053 bytes
Input file size = 13143 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 11400
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 11400

Output IDAT size = 11400 bytes (1653 bytes decrease)
Output file size = 11478 bytes (1665 bytes = 12.67% decrease)

** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/auc_based_dosing_files/figure-html/map_plot_tdm-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 16695 bytes
Input file size = 16797 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 13682
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 13682

Output IDAT size = 13682 bytes (3013 bytes decrease)
Output file size = 13760 bytes (3037 bytes = 18.08% decrease)

--- finished re-building ‘auc_based_dosing.Rmd’

--- re-building ‘classic_posologyr_models.Rmd’ using rmarkdown
--- finished re-building ‘classic_posologyr_models.Rmd’

--- re-building ‘multiple_endpoints.Rmd’ using rmarkdown
** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/multiple_endpoints_files/figure-html/unnamed-chunk-5-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 12529 bytes
Input file size = 12619 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 10623
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 10623

Output IDAT size = 10623 bytes (1906 bytes decrease)
Output file size = 10701 bytes (1918 bytes = 15.20% decrease)

** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/multiple_endpoints_files/figure-html/unnamed-chunk-6-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 11969 bytes
Input file size = 12059 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 9776
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 0		IDAT size = 9776

Output IDAT size = 9776 bytes (2193 bytes decrease)
Output file size = 9854 bytes (2205 bytes = 18.29% decrease)

--- finished re-building ‘multiple_endpoints.Rmd’

--- re-building ‘patient_data.Rmd’ using rmarkdown
--- finished re-building ‘patient_data.Rmd’

--- re-building ‘population_models.Rmd’ using rmarkdown
--- finished re-building ‘population_models.Rmd’

--- re-building ‘route_of_administration.Rmd’ using rmarkdown
** Processing: /data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/vign_test/posologyr/vignettes/route_of_administration_files/figure-html/plot_map-1.png
288x288 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 16935 bytes
Input file size = 17037 bytes

Trying:
  zc = 9  zm = 8  zs = 0  f = 0
  zc = 9  zm = 8  zs = 1  f = 0
  zc = 1  zm = 8  zs = 2  f = 0
  zc = 9  zm = 8  zs = 3  f = 0
  zc = 9  zm = 8  zs = 0  f = 5		IDAT size = 15220
  zc = 9  zm = 8  zs = 1  f = 5
  zc = 1  zm = 8  zs = 2  f = 5
  zc = 9  zm = 8  zs = 3  f = 5
                               
Selecting parameters:
  zc = 9  zm = 8  zs = 0  f = 5		IDAT size = 15220

Output IDAT size = 15220 bytes (1715 bytes decrease)
Output file size = 15298 bytes (1739 bytes = 10.21% decrease)

--- finished re-building ‘route_of_administration.Rmd’

SUMMARY: processing the following file failed:
  ‘a_posteriori_dosing.Rmd’

Error: Vignette re-building failed.
Execution halted

* checking PDF version of manual ... OK
* checking HTML version of manual ... OK
* checking for non-standard things in the check directory ... OK
* checking for detritus in the temp directory ... OK
* DONE

Status: 1 ERROR
See
  ‘/data/gannet/ripley/R/packages/tests-MKL/posologyr.Rcheck/00check.log’
for details.

Command exited with non-zero status 1
Time 13:45.87, 951.74 + 35.84


```

**Comment by levenc** [comment]
In addition, CRAN also reports frequent memory corruption.

**Comment by mattfidler** [comment]
This is probably from `rxode2` and only seems to show up with the 'MKL' build:

```
Tests using R-devel on x86_64 Fedora 34 Linux with alternative BLAS/LAPACK
implementations.  Except as noted below, R was configured as per
https://www.stats.ox.ac.uk/pub/bdr/Rconfig/r-devel-linux-x86_64-fedora-gcc

ATLAS:
Serial ATLAS using the Fedora shared libraries. (Currently version 3.10.3.)
libRblas.so was replaced by a symlink to /usr/lib64/atlas/libsatlas.so .
It reports using LAPACK 3.10.0.

MKL:
Intel MKL '2023.2.0' used to build Rblas and also provides LAPACK
(which is version 3.10.1).  R was configured by (csh script)

setenv MKL_LIB_PATH /usr/local/MKL/mkl/lib/intel64
setenv LD_LIBRARY_PATH $MKL_LIB_PATH
setenv MKL "-L$MKL_LIB_PATH -lmkl_gf_lp64 -lmkl_core -lmkl_sequential"
~/R/svn/R-devel/configure -C --with-blas="$MKL" --with-lapack --enable-lto=R

OpenBLAS:

Serial OpenBLAS using the Fedora shared libraries (currently version 0.3.19).
libRblas.so was replaced by a symlink to /usr/lib64/libopenblas.so (part of
openblas-devel, linked from openblas-serial). This reports using LAPACK 3.9.0.

Pto tem, using

setenv _R_CHECK_MBCS_CONVERSION_FAILURE_ true
```

So you are unlikely to see it elsewhere.

**Comment by levenc** [comment]
Ah, you’re right—thank you for pointing that out. It seems I’ve introduced something unintended in the vignettes since version 1.2.26.

**Comment by levenc** [comment]
Good news—it crashes on my computer too! 😄

It appears that #53 has had some unintended consequences. When trying to update the model (like adding dosing or observations), it seems I may have created a “Frankenmodel”. I’ll work on finding a better solution to fix this issue.

The `a_priori_dosing` vignette failed with the following output:

```r
<div style="position: absolute; overflow: hidden; top: 0px; right: 0px; bottom: 0px; width: 668px;"><div style="width: 668px; height: 806.8px; position: absolute; inset: 0px;"><div style="position: relative; width: 668px; height: 806.8px;"><div style="position: absolute; overflow: hidden; z-index: 0; inset: 0px 0px 351px;"><div role="region" aria-label="Console" class="windowframe GND-IWGDGW" style="-webkit-font-smoothing: subpixel-antialiased; position: absolute; inset: 0px;"><div style="border: 1px solid rgb(214, 218, 220); border-radius: 3px 3px 0px 0px; inset: 3px 3px 5px; overflow: hidden; position: absolute;"><div style="position: absolute; inset: 0px;"><div style="position: absolute; overflow: hidden; inset: 0px;"><div class="gwt-TabLayoutPanel GND-IWGDJS gwt-TabLayoutPanel-Workbench" style="background: rgb(244, 248, 249); position: absolute; inset: 0px; width: 666.4px; height: 454.2px;"><div style="inset: 23px 0px 0px; position: absolute; overflow: hidden;"><div class="gwt-TabLayoutPanelContentContainer" style="position: absolute; inset: 0px;"><div style="position: absolute; overflow: hidden; left: 0px; top: 0px; width: 666.4px; height: 431.2px;"><div role="tabpanel" aria-label="Render" id="rstudio_workbench_panel_render" class="gwt-TabLayoutPanelContent" style="background: rgb(244, 248, 249); position: absolute; inset: 0px;"><div style="position: absolute; overflow: hidden; inset: 0px;"><div style="width: 666.4px; height: 431.2px; position: absolute; inset: 0px;"><div style="position: relative; width: 666.4px; height: 431.2px;"><div style="position: absolute; overflow: hidden; inset: 0px;"><div style="position: absolute; width: 666.4px; height: 431.2px; inset: 0px;"><div style="position: absolute; overflow: hidden; inset: 22px 0px 0px;"><div style="width: 666.4px; height: 409.2px; position: absolute; inset: 0px;"><div style="width: 666.4px; height: 409.2px;"><div class="ace_editor ace_scroller" style="position: relative; overflow: auto; font-style: normal; font-variant: normal; font-weight: normal; font-stretch: normal; font-size: 12px; line-height: normal; font-family: &quot;Ubuntu Mono&quot;, monospace !important; direction: ltr; text-align: left; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); top: 0px; bottom: 0px; background-color: rgb(255, 255, 255); user-select: none; cursor: text; border: none !important; color: rgb(0, 0, 0); zoom: 1; width: 654.4px; height: 409.2px;"><div style="position: relative; zoom: 1;"><pre tabindex="0" class="GND-IWGDH3B GND-IWGDL3B GND-IWGDOW" style="font-family: &quot;Ubuntu Mono&quot;, monospace; font-size: 10pt !important; outline: none; border: none; word-break: break-all; margin: 0px; user-select: text; white-space: pre-wrap !important; box-sizing: border-box; width: 654.4px; padding: 6px; line-height: 1.25;"><span class="GND-IWGDH3B  ace_constant ace_language" style="outline: none; border: none; word-break: break-all; margin: 0px; user-select: text; white-space: pre-wrap !important; color: rgb(57, 148, 106);">*** caught segfault ***
address (nil), cause 'memory not mapped'

Traceback:
 1: rxSolveSEXP(object, .ctl, .nms, .xtra, params, events, inits,     setupOnlyS = .setupOnly)
 2: withCallingHandlers(expr, warning = function(w) {    assign(".ws", unique(c(w$message, .ws)), .thisEnv)})
 3: withCallingHandlers(expr, warning = function(w) if (inherits(w,     classes)) tryInvokeRestart("muffleWarning"))
 4: suppressWarnings(withCallingHandlers(expr, warning = function(w) {    assign(".ws", unique(c(w$message, .ws)), .thisEnv)}))
 5: .collectWarnings(rxSolveSEXP(object, .ctl, .nms, .xtra, params,     events, inits, setupOnlyS = .setupOnly), lst = TRUE)
 6: doTryCatch(return(expr), name, parentenv, handler)
 7: tryCatchOne(expr, names, parentenv, handlers[[1L]])
 8: tryCatchList(expr, classes, parentenv, handlers)
 9: tryCatch({    .envReset$ret &lt;- .collectWarnings(rxSolveSEXP(object, .ctl,         .nms, .xtra, params, events, inits, setupOnlyS = .setupOnly),         lst = TRUE)}, error = function(e) {    if (regexpr("not provided by package", e$message) != -1) {        if (.envReset$cacheReset) {            .malert("unsuccessful cache reset; try manual reset with 'rxClean()'")            stop(e)        }        else {            gc()            .minfo("try resetting cache")            rxode2::rxClean()            .envReset$cacheReset &lt;- TRUE            .envReset$reset &lt;- TRUE            .msuccess("done")        }    }    else if (regexpr("maximal number of DLLs reached", e$message) !=         -1) {        if (.envReset$unload) {            .malert("Could not unload rxode2 models, try restarting R")            stop(e)        }        else {            gc()            .minfo("try resetting cache and unloading all rxode2 models")            try(rxode2::rxUnloadAll())            rxode2::rxClean()            .envReset$unload &lt;- TRUE            .envReset$reset &lt;- TRUE            .msuccess("done")        }    }    else {        stop(e)    }})
10: rxSolve.default(.args.object, events = .newEt, updateObject = TRUE)
11: rxSolve(.args.object, events = .newEt, updateObject = TRUE)
12: (function (...) {    .newEt &lt;- .et    .newEt$clear.sampling()    .newEt$add.sampling(...)    invisible(rxSolve(.args.object, events = .newEt, updateObject = TRUE))})(c(0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 4, 4.1, 4.2, 4.3, 4.4, 4.5, 4.6, 4.7, 4.8, 4.9, 5, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9, 6, 6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9, 7, 7.1, 7.2, 7.3, 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6, 8.7, 8.8, 8.9, 9, 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 9.7, 9.8, 9.9, 10, 10.1, 10.2, 10.3, 10.4, 10.5, 10.6, 10.7, 10.8, 10.9, 11, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 11.7, 11.8, 11.9, 12, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 12.8, 12.9, 13, 13.1, 13.2, 13.3, 13.4, 13.5, 13.6, 13.7, 13.8, 13.9, 14, 14.1, 14.2, 14.3, 14.4, 14.5, 14.6, 14.7, 14.8, 14.9, 15, 15.1, 15.2, 15.3, 15.4, 15.5, 15.6, 15.7, 15.8, 15.9, 16, 16.1, 16.2, 16.3, 16.4, 16.5, 16.6, 16.7, 16.8, 16.9, 17, 17.1, 17.2, 17.3, 17.4, 17.5, 17.6, 17.7, 17.8, 17.9, 18, 18.1, 18.2, 18.3, 18.4, 18.5, 18.6, 18.7, 18.8, 18.9, 19, 19.1, 19.2, 19.3, 19.4, 19.5, 19.6, 19.7, 19.8, 19.9, 20))
13: `$&lt;-.rxSolve`(`*tmp*`, time, value = c(0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 4, 4.1, 4.2, 4.3, 4.4, 4.5, 4.6, 4.7, 4.8, 4.9, 5, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9, 6, 6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9, 7, 7.1, 7.2, 7.3, 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6, 8.7, 8.8, 8.9, 9, 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 9.7, 9.8, 9.9, 10, 10.1, 10.2, 10.3, 10.4, 10.5, 10.6, 10.7, 10.8, 10.9, 11, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 11.7, 11.8, 11.9, 12, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 12.8, 12.9, 13, 13.1, 13.2, 13.3, 13.4, 13.5, 13.6, 13.7, 13.8, 13.9, 14, 14.1, 14.2, 14.3, 14.4, 14.5, 14.6, 14.7, 14.8, 14.9, 15, 15.1, 15.2, 15.3, 15.4, 15.5, 15.6, 15.7, 15.8, 15.9, 16, 16.1, 16.2, 16.3, 16.4, 16.5, 16.6, 16.7, 16.8, 16.9, 17, 17.1, 17.2, 17.3, 17.4, 17.5, 17.6, 17.7, 17.8, 17.9, 18, 18.1, 18.2, 18.3, 18.4, 18.5, 18.6, 18.7, 18.8, 18.9, 19, 19.1, 19.2, 19.3, 19.4, 19.5, 19.6, 19.7, 19.8, 19.9, 20))
14: `$&lt;-`(`*tmp*`, time, value = c(0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 4, 4.1, 4.2, 4.3, 4.4, 4.5, 4.6, 4.7, 4.8, 4.9, 5, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9, 6, 6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9, 7, 7.1, 7.2, 7.3, 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6, 8.7, 8.8, 8.9, 9, 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 9.7, 9.8, 9.9, 10, 10.1, 10.2, 10.3, 10.4, 10.5, 10.6, 10.7, 10.8, 10.9, 11, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 11.7, 11.8, 11.9, 12, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 12.8, 12.9, 13, 13.1, 13.2, 13.3, 13.4, 13.5, 13.6, 13.7, 13.8, 13.9, 14, 14.1, 14.2, 14.3, 14.4, 14.5, 14.6, 14.7, 14.8, 14.9, 15, 15.1, 15.2, 15.3, 15.4, 15.5, 15.6, 15.7, 15.8, 15.9, 16, 16.1, 16.2, 16.3, 16.4, 16.5, 16.6, 16.7, 16.8, 16.9, 17, 17.1, 17.2, 17.3, 17.4, 17.5, 17.6, 17.7, 17.8, 17.9, 18, 18.1, 18.2, 18.3, 18.4, 18.5, 18.6, 18.7, 18.8, 18.9, 19, 19.1, 19.2, 19.3, 19.4, 19.5, 19.6, 19.7, 19.8, 19.9, 20))
15: eval(expr, envir, enclos)
16: eval(expr, envir, enclos)
17: eval_with_user_handlers(expr, envir, enclos, user_handlers)
18: withVisible(eval_with_user_handlers(expr, envir, enclos, user_handlers))
19: withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler)
20: handle(ev &lt;- withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler))
21: timing_fn(handle(ev &lt;- withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler)))
22: evaluate_call(expr, parsed$src[[i]], envir = envir, enclos = enclos,     debug = debug, last = i == length(out), use_try = stop_on_error !=         2L, keep_warning = keep_warning, keep_message = keep_message,     log_echo = log_echo, log_warning = log_warning, output_handler = output_handler,     include_timing = include_timing)
23: evaluate::evaluate(...)
24: evaluate(code, envir = env, new_device = FALSE, keep_warning = if (is.numeric(options$warning)) TRUE else options$warning,     keep_message = if (is.numeric(options$message)) TRUE else options$message,     stop_on_error = if (is.numeric(options$error)) options$error else {        if (options$error &amp;&amp; options$include)             0L        else 2L    }, output_handler = knit_handlers(options$render, options))
25: in_dir(input_dir(), expr)
26: in_input_dir(evaluate(code, envir = env, new_device = FALSE,     keep_warning = if (is.numeric(options$warning)) TRUE else options$warning,     keep_message = if (is.numeric(options$message)) TRUE else options$message,     stop_on_error = if (is.numeric(options$error)) options$error else {        if (options$error &amp;&amp; options$include)             0L        else 2L    }, output_handler = knit_handlers(options$render, options)))
27: eng_r(options)
28: block_exec(params)
29: call_block(x)
30: process_group(group)
31: withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),     error = function(e) if (xfun::pkg_available("rlang", "1.0.0")) rlang::entrace(e))
32: xfun:::handle_error(withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),     error = function(e) if (xfun::pkg_available("rlang", "1.0.0")) rlang::entrace(e)),     function(loc) {        setwd(wd)        write_utf8(res, output %n% stdout())        paste0("\nQuitting from lines ", loc)    }, if (labels[i] != "") sprintf(" [%s]", labels[i]), get_loc)
33: process_file(text, output)
34: knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
35: rmarkdown::render("/home/cyril/Documents/recherche/outils R/posologyr/vignettes/a_priori_dosing.Rmd",     encoding = "UTF-8", output_dir = "/tmp/Rtmps7nNVB/preview-130184a72c582.dir")
An irrecoverable exception occurred. R is aborting now ...
```

**Comment by mattfidler** [comment]
If you have a small reproducible example I can prevent it from happening
again.

**Comment by levenc** [comment]
Thank you for your help. The following example reproduces the error and occasionally crashes the R session. However, it’s not consistent—most of the time, the action completes with a warning: "can not update object."

```r
patient_data <- data.frame(ID=1,
                           TIME=c(0.0,1.0,14.0),
                           DV=c(NA,25.0,5.5),
                           AMT=c(2000,0,0),
                           DUR=c(0.5,NA,NA),
                           EVID=c(1,0,0),
                           CLCREAT=c(80,80,60),
                           WT=65)

mod_run001 <- function() {
  ini({
    THETA_Cl <- 4.0
    THETA_Vc <- 70.0
    THETA_Ka <- 1.0
    ETA_Cl ~ 0.2
    ETA_Vc ~ 0.2
    ETA_Ka ~ 0.2
    prop.sd <- sqrt(0.05)
  })
  model({
    TVCl <- THETA_Cl*CLCREAT/90
    TVVc <- THETA_Vc*WT/70
    TVKa <- THETA_Ka

    Cl <- TVCl*exp(ETA_Cl)
    Vc <- TVVc*exp(ETA_Vc)
    Ka <- TVKa*exp(ETA_Ka)

    K20 <- Cl/Vc
    Cc <- centr/Vc

    d/dt(depot) = -Ka*depot
    d/dt(centr) = Ka*depot - K20*centr
    Cc ~ prop(prop.sd)
  })
}

issue54 <- posologyr::poso_simu_pop(patient_data,mod_run001)

issue54$model$time <- seq(0,20,b=0.1)
```

My goal is to enable the increase of observations or the addition of doses to an rxode2 model that was initially solved using events recorded in a dataframe (rather than an eventTable).

However, using `add.sampling` or `add.dosing` in this context fails if the model requires covariates present in the initial dataframe. To address this, I’ve implemented two workarounds:

1. Passing covariate values as model parameters in `$params`: This approach works, but the covariate values cannot change over time.
2. Creating an eventTable from the dataframe: With this method, the eventTable can be extended with the desired observations and doses, and then used with `rxode2::rxSolve()` alongside the model and parameter values.

Since posologyr version 1.2.26, `poso_simu_pop()` has been generating a model based on the second option. Attempting to modify this model is what’s leading to the warnings and errors encountered.

**Comment by mattfidler** [comment]
Yeah, this isn't tested extensively.  I would do the following:

- For each time-point use `approxfun()` to create a function for the covariate; something like `augPred()` does for fits in `nlmixr2est`: 


https://github.com/nlmixr2/nlmixr2est/blob/3552307e524d4a9ab38530770fca3b6995a1c1f9/R/augPred.R#L65-L69


- Interpolate the covariate based on the new times as above.  You can change this to match `nocb`, `locf`, `linear` and `midpoint`.  These are tested against and shown in the `rxode2` testing; these are in the testing suite here:

https://github.com/nlmixr2/rxode2/blob/f8a519eb013ac1d338cfd1f8e8a3bcc655a588b6/tests/testthat/test-cov.R#L210

**Comment by mattfidler** [comment]
I ran this and I couldn't reproduce it in the latest development version.

Perhaps it is already fixed or for some reason I simply cannot reproduce it

**Comment by levenc** [comment]
Thanks for giving it a try! It might actually be an issue with my machine. After re-reading the CRAN log, I realized the error is coming from the vignette using `poso_estim_map`, not the one with `poso_simu_pop`.

I’ve made two changes:

1. I accepted your pull request to switch the vignettes to articles.
2. I followed your suggestion to extrapolate covariates and created a function, `poso_replace_et()`, to automate this process and simplify things for posologyr users.

I’m going to resubmit it to CRAN—fingers crossed they’ll approve it!
